test/integration_test/s3/getObject.spec.js

```
@@ -0,0 +1,209 @@
+    delectBucket(done);
+  })
+
+  it('should correctly call callback when success', function(done) {
+    s3.getObject({
+      Bucket: params.Bucket,
+      Key: params.Key,
+      Range: 'bytes=0-5',
+    }, function(err, data) {
+      expect(err).to.be['null'];
+      expect(data).to.not.be['null'];
+      expect(data.Body.toString()).to.equal('this i');
+      expect(data.ContentLength).to.equal(6);
+      expect(data.Range).to.equal('bytes 0-5/15')
+      done();
+    });
+  });
+
+  it('should correctly call callback when fail', function(done) {
+    s3.getObject({
+      Bucket: params.Bucket,
+      Key: 'wrongkey',
+    }, function(err, data) {
+      expect(data).to.be['null'];
+      expect(err).to.not.be['null'];
+      expect(err.code).to.equal('NoSuchKey');
+      done();
+    });
+  });
+
+  it('should correctly create read stream', function(done) {
+    var data = '';
+    var error = null;
+    var s = s3.getObject({
+      Bucket: params.Bucket,
+      Key: params.Key,
+    }).createReadStream();
+    s.on('readable', function() {
+      var dat = s.read();
+      if(dat) {
+        data += dat;
+      }
+    })
+    s.on('error', function(err) {
+      error = err;
+    })
+    s.on('end', function() {
+      expect(error).to.be['null'];
+      expect(data).to.not.be['null'];
+      expect(data).to.equal('this is a test!');
+      done();
+    })
+  });
+
+  it('should raise error in creating read stream when fail', function(done) {
+    var data = '';
+    var error = null;
+    var s = s3.getObject({
+      Bucket: params.Bucket,
+      Key: 'wrongKey',
+    }).createReadStream();
+    s.on('readable', function() {
+      var dat = s.read();
+      if(dat) {
+        data += dat;
+      }
+    })
+    s.on('error', function(err) {
+      error = err;
+      expect(error).to.not.be['null'];
+      expect(error.code).to.equal('NoSuchKey');
+      done();
+    })
+    s.on('end', function() {
+      expect(data).to.equal('');
```

Does `end` ever get triggered? If it shouldn't, maybe we should throw an error here. If it should, then `done()` should land here assuming `end` is called after `error`.